momentGW/pbc/ints.py

```diff
@@ -307,6 +312,82 @@ def transform(self, do_Lpq=None, do_Lpx=True, do_Lia=True):
 
         logger.timer(self, "transform", *cput0)
 
+    def input_integrals(self):
+        """
+        Build CDERI elements using THC integrals imported from a h5py
+        file containing a dictionary of components. Inside the dict, a
+        'collocation_matrix' and a 'coulomb_matrix' must be contained
+        with shapes (MO, aux) and (aux,aux) respectively.
+        """
+        if self.input_path is None:
+            raise ValueError(
+                "A file path containing the THC integrals is needed for the THC implementation"
+            )
+        if "thc" not in self.input_path or "THC" not in self.input_path:
+            raise ValueError("File path must contain 'thc' or 'THC' for THC implementation")
+
+        thc_eri = h5py.File(self.input_path, "r")
+
+        kpts_imp = np.array(thc_eri["kpts"])
+
+        if kpts_imp.shape[0] != len(self.kpts):
+            raise ValueError("Number of kpts imported differs from pyscf")
+        if not np.allclose(kpts_imp, self.kpts._kpts) and not np.allclose(
+            kpts_imp, -self.kpts._kpts
+        ):
+            raise ValueError("Different kpts imported to those in pyscf")
+
+        Lpx = {}
+        Lia = {}
+        Lai = {}
+        self._naux = [np.array(thc_eri["coulomb_matrix"])[0, ..., 0].shape[0]] * len(self.kpts)
```

Comment:
Don't assign new variables in functions that aren't initialised in `__init__` -- add `self._naux = None` to the `__init__` (the one for this class, not a subclass, I'd say), then in the `naux` property do:
```
@property
def naux(self):
  if self._rot is None:
    if self._naux is not None:
      return self._naux
...
```

Reply:
Changed this in the original thc implementation so that the new and old method are the same.